### PR TITLE
Added setMultiZoneEffect support

### PR DESCRIPTION
**Pull request by Shakesbeard**
Adds support for starting and stopping the MOVE effect of multizone devices

### CODE REVIEW

**Comment by ristomatti** [comment]
This looks promising, thanks! It might take a moment until we get to test this so please bear with us!

**Review by Sawtaytoes — APPROVED** [review]

**Comment by Sawtaytoes** [comment]
@ristomatti Not seeing anything out of the ordinary here. I like that it specifies not just an effect, but that it's a multi-zone effect.

**Comment by ristomatti** [comment]
Edit: commented on a wrong PR. I'll try to check this soon. Now I have time.

File: example/setMultiZoneEffect.js
Comment (ristomatti) [inline example/setMultiZoneEffect.js]: Use an arrow function here already as we'll move on to those soonish anyway when my PR will get merged. Rename to `onPacketReceived`?

File: example/setMultiZoneEffect.js
Comment (ristomatti) [inline example/setMultiZoneEffect.js]: "Sent"?

File: example/setMultiZoneEffect.js
Comment (ristomatti) [inline example/setMultiZoneEffect.js]: Arrow function here also.

File: example/setMultiZoneEffect.js
Comment (ristomatti) [inline example/setMultiZoneEffect.js]: And here.

File: src/lifx/packets/setMultiZoneEffect.js
Comment (ristomatti) [inline src/lifx/packets/setMultiZoneEffect.js]: How about just:
```js
const {constants, utils} = require('../../lifx');
```

File: src/lifx/packets/setMultiZoneEffect.js
Comment (ristomatti) [inline src/lifx/packets/setMultiZoneEffect.js]: Use `const`.

File: src/lifx/packets/setMultiZoneEffect.js
Comment (ristomatti) [inline src/lifx/packets/setMultiZoneEffect.js]:
```js
const obj = {};
let offset = 0;
```

File: src/lifx/packets/setMultiZoneEffect.js
Comment (ristomatti) [inline src/lifx/packets/setMultiZoneEffect.js:36]: Could something like this help: https://www.npmjs.com/package/node-int64 ?

**Review by ristomatti — COMMENTED** [review]
I'm beginning to get a bit tired but am I correct that this is basically a partial implementation of a feature. Allowing manually creating a packet (normally only used internally by the library) to do this, instead of adding a new method to a `Light` to do this?

File: src/lifx/packets/setMultiZoneEffect.js
Comment (Shakesbeard) [reply src/lifx/packets/setMultiZoneEffect.js:36]: I tried with in64 implementations on my local copy but couldn't get it to work. Hence why I left that out as nodejs is about to add in64 support in the core at some point.

**Comment by Shakesbeard** [comment]
Guess I will have another look at this.

File: example/setMultiZoneEffect.js
Comment (Shakesbeard) [reply example/setMultiZoneEffect.js]: uhm.. yes. You are correct :D

File: example/setMultiZoneEffect.js
Comment (Shakesbeard) [reply example/setMultiZoneEffect.js]: Not sure what you mean. const onPacketReceived = () => {?

Comment (Shakesbeard) [reply example/setMultiZoneEffect.js]: Should be fixed. Please crosscheck.

**Comment by Shakesbeard** [comment]
Also added a prototype on the light object now. Not sure whether this is sufficient. Let me know if you have second thoughts.

**Comment by ristomatti** [comment]
I finally got to test this. The code seems to work but there's some bugs. Variables for constants are wrong for instance. I'll merge this to `set-multizone-effect`, fix the issues and create a new PR when done.

**Comment by Shakesbeard** [comment]
@ristomatti Sorry, but can you help me here? I cannot find any var/let which should be const in the merged commit. Maybe I am blind but could you point out an example where you see that issue?

**Comment by ristomatti** [comment]
@Shakesbeard You can check my changes here: https://github.com/node-lifx/lifx-lan-client/compare/set-multizone-effect (commits 9359951 and 
badb821).

**Comment by ristomatti** [comment]
Sorry I didn't expect for you to respond so fast as this has been sitting here so long. I'm still in the progress of trying to figure some things out. For example what is the purpose of the commented out code in `packets/setMultiZoneEffect.js`?

Most of my changes were related to ESLint errors. You can see them by running `npm run lint`.

**Comment by ristomatti** [comment]
> @ristomatti Sorry, but can you help me here? I cannot find any var/let which should be const in the merged commit. Maybe I am blind but could you point out an example where you see that issue?

Ah now I read this correctly. I didn't mean `var` and `const` but there were references to `lifx_constants` which was supposed to be just `constants` in both the example and `lights.js`.

**Comment by ristomatti** [comment]
Changes will be merged with PR #33 once I've tested them. README.md also needs to be updated before merging.

**Comment by Shakesbeard** [comment]
> Sorry I didn't expect for you to respond so fast as this has been sitting here so long. I'm still in the progress of trying to figure some things out. For example what is the purpose of the commented out code in `packets/setMultiZoneEffect.js`?
> 
> Most of my changes were related to ESLint errors. You can see them by running `npm run lint`.

Oh yes, that commeted out pieces are things which are supposed to be supported in the LIFX protocol, but I disabled those parts due to the lack of proper int64 support. Those code blocks will get relevant once node has proper int64 support in the core. I tested locally with some 3rd party int64 implementation but I could not get those flying with the buffer serialization. So in essence, the commented out code is basically a feature on hold.

**Comment by ristomatti** [comment]
@Shakesbeard Please check if everything looks ok to you at #33.

**Comment by Shakesbeard** [comment]
@ristomatti Looks good! Thanks for the integration help!

**Pull request merged**
